docs/getting-started/dashboard_tutorial.md

```diff
@@ -0,0 +1,1788 @@
+            rx.foreach(
+                State3.users, show_user
+            ),
+        ),
+        variant="surface",
+        size="3",
+    ),
+)
+```
+
+```python
+class State(rx.State):
+    users: list[User] = [
+        User(name="Danilo Sousa", email="danilo@example.com", gender="Male"),
+        User(name="Zahra Ambessa", email="zahra@example.com", gender="Female"),
+    ]
+
+    def add_user(self, form_data: dict):
+        self.users.append(User(**form_data))
+
+
+def show_user(user: User):
+    """Show a person in a table row."""
+    return rx.table.row(
+        rx.table.cell(user.name),
+        rx.table.cell(user.email),
+        rx.table.cell(user.gender),
+    )
+
+def form():
+    return rx.form(
+        rx.vstack(
+            rx.input(
+                placeholder="User Name", name="name", required=True
+            ),
+            rx.input(
+                placeholder="user@reflex.dev",
+                name="email",
+            ),
+            rx.select(
+                ["Male", "Female"],
+                placeholder="Male",
+                name="gender",
+            ),
+            rx.button("Submit", type="submit"),
+        ),
+        on_submit=State.add_user,
+        reset_on_submit=True,
+    )
+
+def index() -> rx.Component:
+    return rx.vstack(
+        form(),
+        rx.table.root(
+            rx.table.header(
+                rx.table.row(
+                    rx.table.column_header_cell("Name"),
+                    rx.table.column_header_cell("Email"),
+                    rx.table.column_header_cell("Gender"),
+                ),
+            ),
+            rx.table.body(
+                rx.foreach(
+                    State.users, show_user
+                ),
+            ),
+            variant="surface",
+            size="3",
+        ),
+    )
+```
+
+
+### Putting the Form in an Overlay
+
+In Reflex we like to make the user interaction as intuitive as possible. Placing the form we just constructed in an overlay creates a focused interaction by dimming the background, and ensures a cleaner layout when you have multiple action points such as editing and deleting as well.
+
+We will place the form inside of a `rx.dialog` component (also called a modal). The `rx.dialog.root` contains all the parts of a dialog, and the `rx.dialog.trigger` wraps the control that will open the dialog. In our case the trigger will be an `rx.button` that says "Add User" as shown below.
+
+```python
+rx.dialog.trigger(
+    rx.button(
+        rx.icon("plus", size=26),
+        rx.text("Add User", size="4"),
+    ),
+)
+```
+
+After the trigger we have the `rx.dialog.content` which contains everything within our dialog, including a title, a description and our form. The key part of the form within the dialog is that it now has two `rx.dialog.close` components. The first is to just close the dialog without submitting the form and he second is to close the dialog and submit the form as shown below.
```

Comment:
Two close components? Why? Ahhh I'm confused now.

Again, maybe first come from the behavior we want (we want two different ways to close it), and then say we should have two close components as a result (instead of the other way around).

Again, if as a user I saw the final product first, it would more naturally make sense.

Reply:
fixed